### DISCUSSION TITLE
Welcome to redisinsight-docker-extension Discussions!

### DISCUSSION THREAD

**Discussion by collabnix** [collabnix/redisinsight-docker-extension#7] [Announcements]
Currently, the extension gets installed on Docker Desktop and one can connect to the Redis database instance without any issue.
But it is not a full-fledge implementation.

Here are the things that goes missing:
- It's always recommended to build UI rather than just embedding the tool into the Extension. Currently this extension doesn't comply with the Docker Extension Style Guide
- Once the extension is up and running, if you try to go to Docker dashboard > Extensions > Click on "I don't have a Redis database" nothing works. Building UI from scratch is the only solution
- All of our extensions must pass the docker extension validate command. Right now I see the output is failing: title "RedisInsight Extension for Docker Desktop" should not include "extension"

```docker extension validate ajeetraina/redisinsight-docker-extension:1.13.0
label "com.docker.extension.additional-urls" is not specified. You can provide more detailed information to users about your extension.
The extension image "ajeetraina/redisinsight-docker-extension:1.13.0" is not valid. see errors: 
- mandatory image label "com.docker.desktop.extension.api.version" is missing
- mandatory image label "org.opencontainers.image.vendor" is missing
- mandatory image label "org.opencontainers.image.description" is missing
- mandatory image label "org.opencontainers.image.title" is missing
- mandatory image label "com.docker.desktop.extension.icon" is missing
- mandatory image label "com.docker.extension.screenshots" is missing
- mandatory image label "com.docker.extension.detailed-description" is missing
- mandatory image label "com.docker.extension.publisher-url" is missing
- mandatory image label "com.docker.extension.changelog" is missing
invalid extension image
```

- Also, this version of RedisInsight is quite old (v1.x). Redis Team suggest to pick up the latest version of RedisInsight that is v2.x

All of these points are being tracked via issues https://github.com/collabnix/redisinsight-docker-extension/issues

**Comment by gagandeepp** [comment]
@ajeetraina I have followed the steps mentioned in the video.Let me know what are next steps to follow.

**Comment by collabnix** [comment]
Here's the plan:
- We need to target RedisInsight v2 instead of v1.0


Important Note: Redis announced a new product called [Redis Stack](https://redis.io/docs/stack/get-started/install/docker/) that comes with RedisInsight. This is available in the form of a Docker Image. One can run the below command to bring up both Redis as well as RedisInsight.

```
 docker run -d --name redis-stack -p 6379:6379 -p 8001:8001 redis/redis-stack:latest
```

So, let's target this.
RedisInsight runs on port 8001 whereas 6379 is the default Redis port.
